internal/sqldb/writer.go

```diff
@@ -281,12 +305,31 @@ func writeTriples(
 	sqlClient *sql.DB,
 	triples []*triple,
 ) error {
+	valueStrings := make([]string, 0, batchSize)
+	valueArgs := make([]interface{}, 0, batchSize*4) // 4 columns to insert
 	for _, t := range triples {
-		sqlStmt := `INSERT INTO triples(subject_id,predicate,object_id,object_value) VALUES (?, ?, ?, ?)`
-		_, err := sqlClient.Exec(sqlStmt, t.subjectID, t.predicate, t.objectID, t.objectValue)
+		sqlValues := "(?, ?, ?, ?)"
+		valueStrings = append(valueStrings, sqlValues)
+		valueArgs = append(valueArgs, t.subjectID, t.predicate, t.objectID, t.objectValue)
+		if len(valueStrings) >= batchSize {
+			sqlStmt := fmt.Sprintf(`INSERT INTO triples(subject_id,predicate,object_id,object_value) VALUES %s`, strings.Join(valueStrings, ","))
+			_, err := sqlClient.Exec(sqlStmt, valueArgs...)
+			if err != nil {
+				return err
+			}
+			log.Printf("[INSERT] triples: %d entries", batchSize)
+			valueStrings = valueStrings[:0] // reset the slices for the next batch
+			valueArgs = valueArgs[:0]
+		}
+	}
+	// Insert remaining data that didn't make a full batch
+	if len(valueStrings) > 0 {
+		sqlStmt := fmt.Sprintf(`INSERT INTO triples(subject_id,predicate,object_id,object_value) VALUES %s`, strings.Join(valueStrings, ","))
```

Comment:
Similar to observations, consider extracting this into a helper function to avoid duplication of code.

Reply:
Done